hawkular-javaagent-itest-parent/hawkular-javaagent-all-itests/hawkular-javaagent-cmdgw-itest/src/test/java/org/hawkular/agent/ws/test/StandaloneDeployApplicationITest.java

```diff
@@ -237,40 +247,49 @@ public void testLocalAndRemoteJMXOperations() throws Throwable {
         // make sure to discover the mbean resources (same MBean discovered by local and remote managed server)
         forceInventoryDiscoveryScan();
 
-        CanonicalPath localPath = testHelper.feedPath(hawkularFeedId).resource("Local JMX~org.hawkular.agent.itest:type=simple").get();
-        Optional<Blueprint> resource = testHelper.getBlueprintFromCP(localPath);
-        Assert.assertTrue(resource.isPresent());
-
-        invokeJMXOperations(localPath);
-
-        CanonicalPath remotePath = testHelper.feedPath(hawkularFeedId).resource("Remote JMX~org.hawkular.agent.itest:type=simple").get();
-        resource = testHelper.getBlueprintFromCP(remotePath);
-        Assert.assertTrue(resource.isPresent());
-
-        invokeJMXOperations(remotePath);
+        Collection<ResourceWithType> mbeans = testHelper.getResourceByType(hawkularFeedId, "Simple ITest MBean", 2);
+        Optional<ResourceWithType> localMbean = mbeans.stream()
+                // TODO [lponce] this is not 100% right, id should be opaque but Local JMX and Remote JMX have same name
```

Comment:
other way, if you want to be less dependent on ID generation, would be to check `.contains("Local JMX")` maybe? (if it doesn't return other positives)

Reply:
Yes, that would be great, but issue is the agent stores for MBeans the same type, on this case it hasn't info about Local JMX or Remote JMX.
I didn't want to dig here so I commented.
In general, I found that perhaps some names could be changed for more easy browsing on this new model.
Anyway, I will try to collect more feedback when I have all these tests up and running, but thx for the comment.
//cc @jmazzitelli